pkg/resources/metrics.go

```diff
@@ -48,11 +51,113 @@ const (
 	alertFor30Mins                    = "30m"
 	alertFor60Mins                    = "60m"
 	alertPercentage                   = "90"
+	//100mb in bytes
+	alertLowStorageThreshold = 100 * 1000 * 1000
 )
 
-// CreatePostgresAvailabilityAlert creates a PrometheusRule alert to watch for the availability
+func ReconcilePostgresAlerts(ctx context.Context, client k8sclient.Client, inst *v1alpha1.RHMI, cr *crov1.Postgres) (v1alpha1.StatusPhase, error) {
+	// create prometheus failed rule
+	_, err := createPostgresResourceStatusPhaseFailedAlert(ctx, client, inst, cr)
+	if err != nil {
+		return v1alpha1.PhaseFailed, fmt.Errorf("failed to create postgres failure alert for %s: %w", cr.Name, err)
+	}
+
+	// create the prometheus deletion rule
+	if _, err = createPostgresResourceDeletionStatusFailedAlert(ctx, client, inst, cr); err != nil {
+		return v1alpha1.PhaseFailed, fmt.Errorf("failed to create postgres deletion prometheus alert for %s: %w", cr.Name, err)
+	}
+
+	if cr.Status.Phase != cro1types.PhaseComplete {
+		return v1alpha1.PhaseAwaitingComponents, nil
+	}
+
+	// create the prometheus pending rule
+	_, err = createPostgresResourceStatusPhasePendingAlert(ctx, client, inst, cr)
+	if err != nil {
+		return v1alpha1.PhaseFailed, fmt.Errorf("failed to create postgres pending alert for %s: %w", cr.Name, err)
```

Comment:
wonder should we just log the error rather that return as may stop other alerts from being created

Reply:
This was moved from the existing code in the reconciler. I think it's usual for the Phase to be returned so that the cr is updated with error. It will continue onto the next stage on the next reconcile if it's able to create the alert..

Reply:
:+1: